source/core/coverage/ut_coverage.pkb

```diff
@@ -98,17 +98,46 @@ create or replace package body ut_coverage is
     return l_result;
   end;
 
+  function is_tmp_table_populated return boolean is
+    l_result integer;
+  begin
+    select 1 into l_result from ut_coverage_sources_tmp where rownum = 1;
+    return (l_result = 1);
+  exception
+    when no_data_found then
+      return false;
+  end;
+
+  procedure populate_tmp_table(a_coverage_options ut_coverage_options, a_skipped_objects ut_object_names) is
+    pragma autonomous_transaction;
+    l_schema_names     ut_varchar2_rows;
+  begin
+    delete from ut_coverage_sources_tmp;
+    l_schema_names := coalesce(a_coverage_options.schema_names, ut_varchar2_rows(sys_context('USERENV','CURRENT_SCHEMA')));
+    if a_coverage_options.file_mappings is not empty then
+      execute immediate get_populate_sources_tmp_sql(a_coverage_options) using a_coverage_options.file_mappings, a_skipped_objects, a_coverage_options.include_objects;
+    else
+      execute immediate get_populate_sources_tmp_sql(a_coverage_options) using l_schema_names, a_skipped_objects, a_coverage_options.include_objects;
+    end if;
+    commit;
+  end;
+
+
   /**
   * Public functions
   */
   procedure coverage_start is
   begin
-    ut_coverage_helper.coverage_start('utPLSQL Code coverage run '||ut_utils.to_string(systimestamp));
+    if not ut_coverage_helper.is_started() then
```

Comment:
we schec the state here and perform the same check in tha package itself.
Maybe we can remove the chack from here and leave it only in `ut_coverage_helper`?